File: examples/scenarios/streak-protection-escalation/snippets/react-native/triggerCancelEvent.ts
// Example placement: success/completion handler for campaign cancellation in JS/TS layer (pseudo-code).
// Adapt to: your app architecture, notification integration, and actual OpenClix runtime API names.
// Shared config: examples/scenarios/streak-protection-escalation/openclix.config.json
// Events used: streak_extended, streak_reset
// Note: OpenClix APIs shown here are placeholders until the reference SDK is published.

type OpenClixManagerLike = {
  ingestEvent: (event: { name: string; appState: 'foreground' | 'background' }) => void;
};

export function triggerStreakProtectionEscalationCancel(openClix: OpenClixManagerLike): void {
  openClix.ingestEvent({
    name: 'streak_extended',
    appState: 'foreground',
  });
  // Alternate cancel path in some apps: streak_reset
}

File: examples/scenarios/interrupted-flow-recovery/snippets/flutter/trigger_cancel_event.dart
// Example placement: success/completion callback for campaign cancellation (pseudo-code).
// Adapt to: your app architecture, notification integration, and actual OpenClix runtime API names.
// Shared config: examples/scenarios/interrupted-flow-recovery/openclix.config.json
// Events used: critical_flow_completed, critical_flow_restarted
// Note: OpenClix APIs shown here are placeholders until the reference SDK is published.

void triggerInterruptedFlowRecoveryCancel(OpenClixManagerLike openClix) {
  final event = OpenClixEventLike(
    name: 'critical_flow_completed',
    appState: OpenClixAppStateLike.foreground,
  );
  openClix.ingestEvent(event);
  // Alternate cancel path in some apps: critical_flow_restarted
}
